packages/expo-modules-core/ios/SharedObjects/SharedObjectRegistry.swift

```diff
@@ -31,28 +31,39 @@ public final class SharedObjectRegistry {
    */
   internal static var pairs = [SharedObjectId: SharedObjectPair]()
 
+  /**
+   The lock queue to keep thread safety for internal data structures.
+   */
+  private static let lockQueue = DispatchQueue(label: "expo.modules.core.SharedObjectRegistry")
```

Comment:
Using a dispatch queue is fine, but I'm wondering if we can use Swift's actors to provide thread safety 🤔 Other functions would have to become `async` and that would be a complex change though.

Reply:
good to learn again from you for the swift actors. a question is that it requires swift 5.5 and we now has swift 5.4. it should be okay since we require xcode 14.3 for react-native 0.73. does bump swift 5.5 make sense to you?

Reply:
Let's keep using the dispatch queue for now. Using actors would require to use the concurrent mode in more places, so I imagine there will be many places to change.

As far as I know, the `swift_version` in the podspec doesn't really matter. It probably uses only the major version (in Xcode's build settings you can only choose between the major versions).

Reply:
cool, thanks Tomasz!